debezium/loadgen/generate_load.py

```diff
@@ -441,104 +446,128 @@ async def run_command_server():
         pass
     log(f"Shutting down command interface.")
 
-##########################
+# We use process-based executors to run purchase and pageview actions;
+# Each has its own kafka producer; purchase action executors also need
+# a database connection.
+producer = None
+cursor = None
+connection = None
+item_prices = None
+def init_resources(kafka_only = False):
+    global producer, cursor, connection, item_prices
+    if not kafka_only:
```

Comment:
There is still the cursor stuff on line 334 that I don't see gated by this.

Reply:
If you look at the initialization code for the executors, the purchases executor is initialized with init_resources(False), which means it will have a cursor.  Line 334 using the global cursor is the purchases loop.  So I believe this is fine; also, from output running there are no exceptions showing up for `cursor` being `None`.